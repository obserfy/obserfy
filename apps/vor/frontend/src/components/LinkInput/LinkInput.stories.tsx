import { FC } from "react"
import { action } from "@storybook/addon-actions"
import LinkInput from "./LinkInput"

export default {
  title: "Core/LinkInput",
  component: LinkInput,
  parameters: {
    componentSubtitle: "Just a simple LinkInput",
  },
}

export const Basic: FC = () => (
  <LinkInput
    isLoading={false}
    onChange={action("onChange")}
    onSave={action("onSave")}
  />
)
