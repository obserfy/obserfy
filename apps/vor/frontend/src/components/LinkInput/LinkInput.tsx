import { Trans } from "@lingui/macro"
import { FC } from "react"
import { Button, Flex, ThemeUIStyleObject } from "theme-ui"
import Input from "../Input/Input"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"

const LinkInput: FC<{
  value?: string
  onChange: (value: string) => void
  onSave: () => void
  isLoading?: boolean
  containerSx?: ThemeUIStyleObject
  inputSx?: ThemeUIStyleObject
}> = ({ inputSx, containerSx, value, onChange, onSave, isLoading }) => (
  <Flex sx={containerSx}>
    <Input
      aria-label="URL"
      sx={{ width: "100%", ...inputSx }}
      p={2}
      placeholder="https://...."
      value={value}
      onChange={(e) => onChange(e.target.value)}
    />
    <Button
      data-cy="add-link"
      ml={2}
      variant="outline"
      disabled={value === "" || isLoading}
      onClick={onSave}
    >
      {isLoading ? <LoadingIndicator /> : <Trans>Add</Trans>}
    </Button>
  </Flex>
)

export default LinkInput
