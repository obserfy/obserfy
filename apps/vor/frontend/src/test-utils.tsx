/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
