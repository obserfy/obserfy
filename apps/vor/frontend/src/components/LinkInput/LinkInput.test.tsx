import { render } from "../../test-utils"
import { Basic } from "./LinkInput.stories"

describe("LinkInput", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
