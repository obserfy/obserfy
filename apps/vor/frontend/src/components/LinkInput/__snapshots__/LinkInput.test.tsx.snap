// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkInput should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  padding: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-4:autofill,
.emotion-4:autofill:hover,
.emotion-4:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-4:autofill:first-line,
.emotion-4:autofill:hover:first-line,
.emotion-4:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-4:-webkit-autofill,
.emotion-4:-webkit-autofill:hover,
.emotion-4:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-4:-webkit-autofill:first-line,
.emotion-4:-webkit-autofill:hover:first-line,
.emotion-4:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-4:hover,
.emotion-4:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-4::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-4::-moz-placeholder {
  font-style: italic;
}

.emotion-4:-ms-input-placeholder {
  font-style: italic;
}

.emotion-4::placeholder {
  font-style: italic;
}

.emotion-4:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: 8px;
}

.emotion-5:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5:hover:disabled,
.emotion-5:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <input
            aria-label="URL"
            class="emotion-4"
            placeholder="https://...."
            value=""
          />
        </div>
      </label>
      <button
        class="emotion-5"
        data-cy="add-link"
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
